### PR TITLE
Blog App - Ruby on Rails: Integration test and fix n+1 problems

### PR DESCRIPTION
**Pull request by lenkon**
#### The following are implemented:
- Solved N+1 problem
- Integration test for the user's index view
- Integration test for the user's show view
- Integration test for the post's index view
- Integration test for the post's show view

#### App's logs screenshots 
- Before N+1 problem is fixed
before fix

- After N+1 problem is fixed
after fix

### CODE REVIEW

File: spec/controllers/views/posts/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    it 'Can see how many likes a post has.' do
+      visit user_posts_path(@user.id)
+      expect(page).to have_content('Likes: 1')
+    end
+
+    it 'Redirects to posts show page' do
+      visit user_posts_path(@user.id)
+      click_link 'Post1'
+      expect(page).to have_current_path user_post_path(@user.id, @post_one.id)
+    end
```
Comment (mikethreels) [inline spec/controllers/views/posts/post_index_spec.rb]: - [ ] According to the requirements there should be a test that asserts you can see a section for pagination if there are more posts than fit on the view.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Inoulen, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job writing the tests

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/views/posts/post_index_spec.rb (hunk 1 above)
Comment (lenkon) [reply spec/controllers/views/posts/post_index_spec.rb]: Hi @mikethreels ,

Thanks for the feedback. I have made the necessary changes.

Best Regards,
K. Inoulen Shanongba

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @lenkon👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with integration specs. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**